### YamlMime:ManagedReference
items:
- uid: System.ArithmeticException
  id: ArithmeticException
  children:
  - System.ArithmeticException.#ctor
  - System.ArithmeticException.#ctor(System.String)
  - System.ArithmeticException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.ArithmeticException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ArithmeticException
  nameWithType: ArithmeticException
  fullName: System.ArithmeticException
  type: Class
  summary: The exception that is thrown for errors in an arithmetic, casting, or conversion operation.
  remarks: "ArithmeticException is the base class for the following exceptions:  \n  \n-   <xref:System.DivideByZeroException>, which is thrown in integer division when the divisor is 0.  For example, attempting to divide 10 by 0 throws a <xref:System.DivideByZeroException> exception.  \n  \n-   <xref:System.NotFiniteNumberException>, which is thrown when an operation is performed on or returns <xref:System.Double.NaN?displayProperty=fullName>, <xref:System.Double.NegativeInfinity?displayProperty=fullName>, <xref:System.Double.PositiveInfinity?displayProperty=fullName>, <xref:System.Single.NaN?displayProperty=fullName>, <xref:System.Single.NegativeInfinity?displayProperty=fullName>, <xref:System.Single.PositiveInfinity?displayProperty=fullName>, and the programming language used does not support those values.  \n  \n-   <xref:System.OverflowException>, which is thrown when the result of an operation is outside the bounds of the target data type. That is, it is less than a number's `MinValue` property or greater than its `MaxValue` property.  For example, attempting to assign 200 + 200 to a <xref:System.Byte> value throws an <xref:System.OverflowException> exception, since 400 greater than 256, the upper bound of the <xref:System.Byte> data type.  \n  \n Your code should not handle or throw this exception. Instead, you should either handle or throw one of its derived classes, since it more precisely indicates the exact nature of the error.  \n  \n For a list of initial property values for an instance of ArithmeticException, see the ArithmeticException constructors.  \n  \n ArithmeticException uses the HRESULT COR_E_ARITHMETIC, which has the value 0x80070216."
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ArithmeticException : SystemException
  inheritance:
  - System.Object
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers:
  - System.Exception.Data
  - System.Exception.GetBaseException
  - System.Exception.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Exception.GetType
  - System.Exception.HelpLink
  - System.Exception.HResult
  - System.Exception.InnerException
  - System.Exception.Message
  - System.Exception.SerializeObjectState
  - System.Exception.Source
  - System.Exception.StackTrace
  - System.Exception.TargetSite
  - System.Exception.ToString
  - System.Object.Equals(System.Object)
  - System.Object.Equals(System.Object,System.Object)
  - System.Object.GetHashCode
  - System.Object.MemberwiseClone
  - System.Object.ReferenceEquals(System.Object,System.Object)
  platform:
  - net-11
  - net-20
  - netcore-10
- uid: System.ArithmeticException.#ctor
  id: '#ctor'
  parent: System.ArithmeticException
  langs:
  - csharp
  name: ArithmeticException()
  nameWithType: ArithmeticException.ArithmeticException()
  fullName: System.ArithmeticException.ArithmeticException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: Initializes a new instance of the <see cref="T:System.ArithmeticException"> class.
  remarks: "This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as \"The arithmetic operation is not allowed.\" This message takes into account the current system culture.  \n  \n The following table shows initial property values for an instance of <xref:System.ArithmeticException>.  \n  \n|Property|Value|  \n|--------------|-----------|  \n|<xref:System.Exception.InnerException%2A>|`null`.|  \n|<xref:System.Exception.Message%2A>|The localized error message string.|"
  syntax:
    content: public ArithmeticException ();
    parameters: []
  platform:
  - net-11
  - net-20
  - netcore-10
- uid: System.ArithmeticException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.ArithmeticException
  langs:
  - csharp
  name: ArithmeticException(String)
  nameWithType: ArithmeticException.ArithmeticException(String)
  fullName: System.ArithmeticException.ArithmeticException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: Initializes a new instance of the <see cref="T:System.ArithmeticException"> class with a specified error message.
  remarks: "The content of the  parameter is intended to be understood by humans. The caller of this constructor is required to ensure that this string has been localized for the current system culture.  \n  \n The following table shows initial property values for an instance of <xref:System.ArithmeticException>.  \n  \n|Property|Value|  \n|--------------|-----------|  \n|<xref:System.Exception.InnerException%2A>|The null reference (`Nothing` in Visual Basic).|  \n|<xref:System.Exception.Message%2A>|The error message string.|"
  syntax:
    content: public ArithmeticException (string message);
    parameters:
    - id: message
      type: System.String
      description: A <see cref="T:System.String"> that describes the error.
  platform:
  - net-11
  - net-20
  - netcore-10
- uid: System.ArithmeticException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.ArithmeticException
  langs:
  - csharp
  name: ArithmeticException(SerializationInfo,StreamingContext)
  nameWithType: ArithmeticException.ArithmeticException(SerializationInfo,StreamingContext)
  fullName: System.ArithmeticException.ArithmeticException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: Initializes a new instance of the <see cref="T:System.ArithmeticException"> class with serialized data.
  remarks: This constructor is called during deserialization to reconstitute the exception object transmitted over a stream. For more information, see [XML and SOAP Serialization](../Topic/XML%20and%20SOAP%20Serialization.md).
  syntax:
    content: protected ArithmeticException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: The object that holds the serialized object data.
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: The contextual information about the source or destination.
  platform:
  - net-11
  - net-20
  - netcore-10
- uid: System.ArithmeticException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.ArithmeticException
  langs:
  - csharp
  name: ArithmeticException(String,Exception)
  nameWithType: ArithmeticException.ArithmeticException(String,Exception)
  fullName: System.ArithmeticException.ArithmeticException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System
  summary: Initializes a new instance of the <see cref="T:System.ArithmeticException"> class with a specified error message and a reference to the inner exception that is the cause of this exception.
  remarks: "An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property. The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or a null reference (`Nothing` in Visual Basic) if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.  \n  \n The following table shows initial property values for an instance of <xref:System.ArithmeticException>.  \n  \n|Property|Value|  \n|--------------|-----------|  \n|<xref:System.Exception.InnerException%2A>|The inner exception reference.|  \n|<xref:System.Exception.Message%2A>|The error message string.|"
  syntax:
    content: public ArithmeticException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: The error message that explains the reason for the exception.
    - id: innerException
      type: System.Exception
      description: The exception that is the cause of the current exception. If the  parameter is not a null reference, the current exception is raised in a `catch` block that handles the inner exception.
  platform:
  - net-11
  - net-20
  - netcore-10
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.ArithmeticException.#ctor
  parent: System.ArithmeticException
  isExternal: false
  name: ArithmeticException()
  nameWithType: ArithmeticException.ArithmeticException()
  fullName: System.ArithmeticException.ArithmeticException()
- uid: System.ArithmeticException.#ctor(System.String)
  parent: System.ArithmeticException
  isExternal: false
  name: ArithmeticException(String)
  nameWithType: ArithmeticException.ArithmeticException(String)
  fullName: System.ArithmeticException.ArithmeticException(String)
- uid: System.String
  parent: System
  isExternal: false
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.ArithmeticException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.ArithmeticException
  isExternal: false
  name: ArithmeticException(SerializationInfo,StreamingContext)
  nameWithType: ArithmeticException.ArithmeticException(SerializationInfo,StreamingContext)
  fullName: System.ArithmeticException.ArithmeticException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: false
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.ArithmeticException.#ctor(System.String,System.Exception)
  parent: System.ArithmeticException
  isExternal: false
  name: ArithmeticException(String,Exception)
  nameWithType: ArithmeticException.ArithmeticException(String,Exception)
  fullName: System.ArithmeticException.ArithmeticException(String,Exception)
- uid: System.String
  parent: System
  isExternal: false
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
